spark/src/main/java/org/apache/iceberg/spark/source/StructInternalRow.java

```diff
@@ -115,12 +120,30 @@ public short getShort(int ordinal) {
 
   @Override
   public int getInt(int ordinal) {
-    return struct.get(ordinal, Integer.class);
+    Object integer = struct.get(ordinal, Object.class);
```

Comment:
The reason for changing this class is discussed in [this thread](https://github.com/apache/iceberg/pull/1790#discussion_r528432217)